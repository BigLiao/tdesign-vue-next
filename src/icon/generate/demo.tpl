<template>
  <!--
    该文件有脚本 tdesign/tdesign-web-vue/src/icon/generate/index.js 自动生成。
    若需要编辑，请编辑源文件：tdesign/tdesign-web-vue/src/icon/generate/demo.tpl 。
  -->
  <div class="t-demo-iconfont">
    <p>目前设计师给到的全部图标</p><br>
    <%= icons %>
  </div>
</template>

<style lang="less">
  .t-demo-iconfont {
    .t-demo-icon {
      width: 130px;
      height: 100px;
      margin-right: 24px;
      display: inline-block;
      text-align: center;
      vertical-align: top;
      .t-icon {
        font-size: 25px;
        margin-right: 0;
        color: #999999;
      }
      .t-icon:hover {
        color: #0052d9;
        font-size: 30px;
      }
      .t-demo-icon-name {
        text-align: center;
        margin-top: 8px;
        font-size: 12px;
      }
    }
  }
</style>
