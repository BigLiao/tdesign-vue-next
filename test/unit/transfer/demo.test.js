import { mount } from '@vue/test-utils';
import demo from '@/examples/transfer/demos/base.vue';

// unit test for component in examples.
describe('Transfer', () => {
  it('base demo works fine', () => {
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
});
