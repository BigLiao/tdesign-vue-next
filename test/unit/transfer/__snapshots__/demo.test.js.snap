// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transfer base demo works fine 1`] = `
<div
  class="t-transfer"
>
  <div
    class="t-transfer-list t-transfer-list-source"
    data="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
    directions="both"
    targetorder="original"
    targetvalue=""
    titles="源列表,目标列表"
  >
    <div
      class="t-transfer-list__header"
    >
      <div>
        <label
          class="t-checkbox"
        >
          <input
            class="t-checkbox__former"
            name=""
            type="checkbox"
            value=""
          />
          <span
            class="t-checkbox__input"
          />
          <span
            class="t-checkbox__label"
          />
        </label>
         
        <span>
          0 / 20项
        </span>
      </div>
       
      <span>
        源列表
      </span>
    </div>
     
    <div
      class="t-transfer-list__body"
    >
      <!---->
       
      <div
        class="t-transfer-list__content"
      >
        <div
          class="t-checkbox-group"
        >
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox t-is-disabled"
            >
              <input
                class="t-checkbox__former"
                disabled=""
                name=""
                type="checkbox"
                value="0"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
             
            <span>
              内容1
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="1"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
             
            <span>
              内容2
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="2"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
             
            <span>
              内容3
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="3"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
             
            <span>
              内容4
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox t-is-disabled"
            >
              <input
                class="t-checkbox__former"
                disabled=""
                name=""
                type="checkbox"
                value="4"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
             
            <span>
              内容5
            </span>
          </li>
        </div>
      </div>
       
      <div
        class="t-transfer-empty"
        style="display: none;"
      >
        暂无数据
      </div>
    </div>
     
    <!---->
     
    <!---->
  </div>
   
  <div
    class="t-transfer-operations"
  >
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-is-disabled t-button--icon-only"
      disabled="disabled"
      type="button"
    >
      <svg
        class="t-icon t-icon-chevron-right"
      >
        <use
          xlink:href="#t-icon-chevron-right"
        />
      </svg>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-is-disabled t-button--icon-only"
      disabled="disabled"
      type="button"
    >
      <svg
        class="t-icon t-icon-chevron-left"
      >
        <use
          xlink:href="#t-icon-chevron-left"
        />
      </svg>
    </button>
  </div>
   
  <div
    class="t-transfer-list t-transfer-list-target"
    data="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
    directions="both"
    targetorder="original"
    targetvalue=""
    titles="源列表,目标列表"
  >
    <div
      class="t-transfer-list__header"
    >
      <div>
        <label
          class="t-checkbox"
        >
          <input
            class="t-checkbox__former"
            name=""
            type="checkbox"
            value=""
          />
          <span
            class="t-checkbox__input"
          />
          <span
            class="t-checkbox__label"
          />
        </label>
         
        <span>
          0 / 0项
        </span>
      </div>
       
      <span>
        目标列表
      </span>
    </div>
     
    <div
      class="t-transfer-list__body"
    >
      <!---->
       
      <div
        class="t-transfer-list__content"
        style="display: none;"
      >
        <div
          class="t-checkbox-group"
        />
      </div>
       
      <div
        class="t-transfer-empty"
      >
        暂无数据
      </div>
    </div>
     
    <!---->
     
    <!---->
  </div>
</div>
`;
